### PR TITLE
Fix missed demangles

### LINKED ISSUE #6348: [BUG]: Not all identifiers are demangled
### Describe the bug

Some identifiers are not demangled, for example
```asm
        jmp     qword ptr [rip + _ZN4core3fmt3num3imp54_$LT$impl$u20$core..fmt..Display$u20$for$u20$usize$GT$3fmt17h7bbbd896a38dcccaE@GOTPCREL]
        mov     edx, OFFSET FLAT:_ZN1a1gEi
```

### Steps to reproduce

Write some code that (for example) is compiled to a `rip`-relative jump: https://godbolt.org/z/5zWGxhxKq

### Expected behavior

The identifiers should be demangled.

I believe these regexes are too restrictive?
https://github.com/compiler-explorer/compiler-explorer/blob/a47d3c68fc98b5c3a17298aa121e13665bb7245a/lib/demangler/base.ts#L47
https://github.com/compiler-explorer/compiler-explorer/blob/a47d3c68fc98b5c3a17298aa121e13665bb7245a/lib/demangler/base.ts#L55

### Reproduction link

https://godbolt.org/z/5zWGxhxKq

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
This PR fixes missed demangle cases pointed out in #6348, namely of the form `mov eax, OFFSET FLAT:_Z...` and `jmp qword ptr [rip + _Z...]`.

Closes #6348.

### CODE REVIEW

File: lib/demangler/base.ts [hunk 1]
```diff
@@ -117,6 +118,7 @@ export class BaseDemangler extends AsmRegex {
 
     protected collectLabels() {
         const symbolMatchers = [
+            this.ptrOffset, // needs to come before the jmp test since the jumpDef regex doesn't match [rip + _Z...]
```
Comment (jeremy-rifkin) [inline lib/demangler/base.ts:121]: I'm not in love with the subtle order of checks here, but it works

**Review by partouf — APPROVED** [review]
The regexes feel a bit weird to me, but i guess it makes sense, and no other tests broke, so, lgtm

**Comment by jeremy-rifkin** [comment]
Thanks for the review. In general this system feels a bit fickle to me and I wonder if we should try to relax things much more and match `\b_[\w$.@]*\b` anywhere that is not a string.

**Comment by partouf** [comment]
this is now live

**Pull request merged**